Text Input Component

BraidenCutforth opened the pull request.

BraidenCutforth commented:
Fixes https://github.com/VikeLabs/Passr/issues/28

daelynj commented on src/services/storage/mock.ts:
```diff
@@ -22,6 +28,10 @@ export const course1: Course = {
     destiredGrade: 0.8,
     items: course1items,
+    dateRange: {
+        start: new Date(thisSemesterStart),
```
@BraidenCutforth I'm curious how nesting a `Date` object in `new Date(_DateObject_)` works?

BraidenCutforth replied:
This creates a deep copy of the date. Want to make sure we aren't manipulating the same date object.

daelynj commented on src/services/storage/mock.ts:
```diff
@@ -53,11 +67,45 @@ const course3: Course = {
+const course4: Course = {
+    name: 'EPHE 155',
+    destiredGrade: 0.9,
+    items: [
```
Let's stick with a pattern - define the items list outside like we did for `course1items` or nest the data like we are here.

daelynj commented on src/services/storage/mock.ts:
```diff
@@ -53,11 +67,45 @@ const course3: Course = {
 }
 
 export const user1: User = {
     name: 'Burt Macklin',
```
> name: 'Burt Macklin'

🤣

BraidenCutforth committed:
Text Input Component

daelynj reviewed (approved):
👍 Nice.

Check out [PropTypes](https://reactjs.org/docs/typechecking-with-proptypes.html). I've definitely been finding this super helpful trying to understand why a prop exists and what it's for.

`TextInput({ prop, prop, prop }: Props)}` really emphasizes the importance of good variable names if you don't use something like `prop-types`.

BraidenCutforth commented:
> 👍 Nice.
> 
> Check out [PropTypes](https://reactjs.org/docs/typechecking-with-proptypes.html). I've definitely been finding this super helpful trying to understand why a prop exists and what it's for.
> 
> `TextInput({ prop, prop, prop }: Props)}` really emphasizes the importance of good variable names if you don't use something like `prop-types`.

I thought prop types was for javascript only but I am interested. I was trying to think of a way to describe the use of the different props

BraidenCutforth committed:
PAS-45: Review Additions

- Add documentation to TextInput props
- Add onBlur function
- Add prop defaults
- Add better error outline

daelynj commented on src/components/TextInput.tsx:
```diff
@@ -34,20 +34,42 @@ const Input = styled.input<{ error: boolean }>`
-function TextInput({ error, value, onChange, label, required, type }: Props) {
+/**
+ *
+ * @param error Display error markers (default: false)
```
Doing this isn't a bad idea either!
